File: Ruby/009.rb
#!/usr/bin/env ruby

target = 1000
limit = Math.sqrt(target)
1.step(limit, 2) do |i|
  (i + 1).step(limit, 2) do |j|
    a, b, c = j**2 - i**2, 2 * i * j, i**2 + j**2
    if a + b + c == target
      puts a * b * c
      break
    end
  end
end

File: Python/009.py
#!/usr/bin/env python
# encoding: utf-8
from math import sqrt


def main():
    limit = int(sqrt(1000))
    for i in range(1, limit, 2):
        for j in range(i+1, limit, 2):
            a, b, c = j * j - i * i, 2 * i * j, i * i + j * j
            if a + b + c == 1000:
                print(a * b * c)
                break


if __name__ == '__main__':
    main()
